datasets/ge/company_registry/crawler.py

```diff
@@ -0,0 +1,172 @@
+    entity.add("name", name)
+    entity.add("classification", legal_form)
+    if reg_date != "NULL":
+        h.apply_date(entity, "incorporationDate", reg_date)
+    for add in h.multi_split(row.pop("address"), ["; ", " / "]):
+        entity.add("address", add)
+    entity.add("status", row.pop("status"))
+    emails = email.replace(" ", "").strip()
+    for email in h.multi_split(emails, [";", ","]):
+        email_clean = registry.email.clean(email)
+        if email_clean is not None:
+            entity.add("email", email)
+    context.emit(entity)
+
+    if director != "NULL":
+        emit_rel(
+            context,
+            "Directorship",
+            row,
+            director,
+            entity,
+            "director_id",
+            "director_citizenship",
+            "director_start_date",
+            "director",
+        )
+        context.audit_data(
+            row,
+            ignore=[
+                "partner_id",
+                "partner_citizenship",
+                "partner_start_date",
+                "partner_share",
+            ],
+        )
+    if partner != "NULL":
+        emit_rel(
+            context,
+            "Ownership",
+            row,
+            partner,
+            entity,
+            "partner_id",
+            "partner_citizenship",
+            "partner_start_date",
+            "owner",
+        )
+        context.audit_data(
+            row,
+            ignore=[
+                "director_id",
+                "director_citizenship",
+                "director_start_date",
+            ],
+        )
+
+
+def emit_rel(
+    context: Context,
+    type_label: str,
+    row: list,
+    name: str,
+    entity,
+    id_key: str,
+    citizenship_key: str,
+    start_date_key: str,
+    relationship_type: str,
+):
+    """Generalized function to process a director or partner."""
+    person_id = row.pop(id_key)
+    person_citizenship = row.pop(citizenship_key)
+    person_start_date = row.pop(start_date_key)
+
+    person = context.make("Person")
+    person.id = context.make_id(name, person_id)
+    person.add("name", name)
+    if person_citizenship != "NULL":
+        for citizenship in h.multi_split(person_citizenship, [","]):
+            person.add("citizenship", citizenship)
+    context.emit(person)
+
+    relationship = context.make(type_label)
+    relationship.id = context.make_id(person.id, relationship_type, entity.id)
+    if person_start_date != "NULL":
+        h.apply_date(relationship, "startDate", person_start_date)
+    relationship.add(relationship_type, person)
+    relationship.add(
+        "organization" if relationship_type == "director" else "asset", entity
+    )
+
+    if relationship_type == "owner":
+        relationship.add("percentage", row.pop("partner_share"))
+
+    context.emit(relationship)
+
+
+def crawl(context: Context) -> None:
+    local_path = context.get_resource_path("companyinfo_v3.csv")
```

Comment:
For my understanding (this is a question for @jbothma maybe): What happens to the resources? Do they get archived somewhere? Because here we're overwriting this file continously.

If this path is not special in any way, then maybe we should have a context processor of the shape of:

```python
with InternalDataBlobAsLocalFile(blob, delete_after_use=True) as local_filepath:
   ...
```

or something like that. For elegance and all. (Should not be part of this PR of course)